### PR TITLE
feat: Add reset password functionality for user accounts

### PR DESCRIPTION
**Pull request by andremugabo**
- Introduce reset password link on login page
- Implement token validation and password reset page
- Ensure input validation for new passwords
- Document code using JSDOC for clear documentation.

[Delivers #187364807]

### CODE REVIEW

File: src/utils/validators.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/**
+ * Validates an email address using a regular expression.
+ * @param {string} email - The email address to validate.
+ * @returns {boolean} Returns true if the email address is valid, false otherwise.
+ */
```
Comment (shebz2023) [inline src/utils/validators.ts:9]: import { z } from 'zod';

File: src/utils/validators.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+export const validateEmail = (email: string): boolean => {
+  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+  return emailRegex.test(email);
+};
+/**
```
Comment (shebz2023) [inline src/utils/validators.ts:21]: const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
export const emailSchema = z.object({
  email: z
    .string()
    .min(1, { message: 'Email is required' })
    .regex(emailRegex, { message: 'Invalid email format' }),
});

**Review by shebz2023 — COMMENTED** [review]
Great work @andremugabo ,lets validate with 'z'

**Review by shebz2023 — DISMISSED** [review]
LGTM

**Review by shebz2023 — DISMISSED** [review]
LGTM

**Review by Oliviier-dev — DISMISSED** [review]
LGTM

File: src/routes/routes.tsx [hunk 3]
```diff
@@ -24,7 +24,8 @@ export const router = createBrowserRouter([
     element: <AuthLayout />,
     children: [
       { path: 'login', index: true, element: <LoginPage /> },
-      { path: 'signup', element: <SignupPage /> },
+      { path: 'reset-password/:token', element: <ResetPassword /> }, // Add route parameter for reset password token
```
Comment (SergeInGithub) [inline src/routes/routes.tsx]:
```suggestion
      { path: 'reset-password/:token', element: <ResetPassword /> },
```

File: src/pages/authpages/ResetPassword.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  const handleGoBack = () => {
+    navigate('/auth/forgotPassword');
+  };
+
+  return (
+    <div className="flex bg-white">
+      <div
+        className="hidden lg:block lg:w-1/2 bg-cover bg-center h-screen"
+        style={{ backgroundImage: `url(${ResetPasswordImage})` }}
+      ></div>
```
Comment (SergeInGithub) [inline src/pages/authpages/ResetPassword.tsx]:
```suggestion
      <div
        className="hidden lg:block lg:w-1/2 bg-cover bg-center h-screen"
        style={{ backgroundImage: `url(${ResetPasswordImage})` }}
      />
```

File: src/pages/authpages/ForgotPassword.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  const handleGoBack = () => {
+    navigate('/auth/login');
+  };
+
+  return (
+    <div className="flex bg-white">
+      <div
+        className="hidden lg:block lg:w-1/2 bg-cover bg-center h-screen"
+        style={{ backgroundImage: `url(${ForgotPasswordImage})` }}
+      ></div>
```
Comment (SergeInGithub) [inline src/pages/authpages/ForgotPassword.tsx]:
```suggestion
      <div
        className="hidden lg:block lg:w-1/2 bg-cover bg-center h-screen"
        style={{ backgroundImage: `url(${ForgotPasswordImage})` }}
      />
```

**Review by SergeInGithub — CHANGES_REQUESTED** [review]
Some reviews below @andremugabo

**Review by SergeInGithub — APPROVED** [review]
LGTM

**Pull request merged**